### PR TITLE
Anchor/Unanchor Smartfridges

### PR DESCRIPTION
**Pull request by Doxxmedearly**
You can move these now, hooray. They won't use power while unanchored, and the sprite will reflect it. You can't use them while they aren't anchored down (Acts like they're unpowered). 

Also, if they are electrified, they will not shock you if unpowered. Hard to be shocked by something that's not on. 

Also cleaned up the code regarding smartfridges checking if they can hold an item or not. Just made lists of accepted items instead of a million messy procs.

### CODE REVIEW

**Comment by Doxxmedearly** [comment]
!review

File: code/modules/cooking/machinery/smartfridge.dm [hunk 1]
```diff
@@ -236,8 +210,14 @@
 	return
 
 /obj/machinery/smartfridge/machinery_process()
-	if(stat & (BROKEN|NOPOWER))
+	update_icon()
```
Comment (alsoandanswer) [inline code/modules/cooking/machinery/smartfridge.dm:213]: I think you could just move the two update_icon() procs into update_use_power instead.

Comment (Doxxmedearly) [reply code/modules/cooking/machinery/smartfridge.dm]: Good call. Done.

File: code/modules/cooking/machinery/smartfridge.dm [hunk 2]
```diff
@@ -409,6 +400,8 @@
 	return data
 
 /obj/machinery/smartfridge/Topic(href, href_list)
+	if(stat & (NOPOWER|BROKEN) || !anchored)
```
Comment (mikomyazaki) [inline code/modules/cooking/machinery/smartfridge.dm:403]: This stat check is already handled by the parent via `/obj/machinery/CanUseTopic`

**Review by mikomyazaki — CHANGES_REQUESTED** [review]

File: code/modules/cooking/machinery/smartfridge.dm (hunk 2 above)
Comment (Doxxmedearly) [reply code/modules/cooking/machinery/smartfridge.dm]: Yeah, should have checked the parent given how jank smartfridge code is. This actually made me look over how I was handling it as a whole, and now it follows how vending machines did it, with unanchoring actually giving it the NOPOWER flag, and using power_change() to update.

Works better with these changes, thank you.

**Review by mikomyazaki — APPROVED** [review]

**Review by JohnWildkins — APPROVED** [review]

**Pull request merged**